testing/scenario_app/ios/Scenarios/Scenarios/AppDelegate.m

```diff
@@ -71,22 +69,24 @@ - (BOOL)application:(UIApplication*)application
   return [super application:application didFinishLaunchingWithOptions:launchOptions];
 }
 
-- (void)readyContextForPlatformViewTests:(NSString*)scenarioIdentifier {
-  FlutterEngine* engine = [[FlutterEngine alloc] initWithName:@"PlatformViewTest" project:nil];
-  [engine runWithEntrypoint:nil];
-
-  FlutterViewController* flutterViewController;
+- (FlutterViewController*)flutterViewControllerForTest:(NSString*)scenarioIdentifier
+                                            withEngine:(FlutterEngine*)engine {
   if ([scenarioIdentifier isEqualToString:@"tap_status_bar"]) {
-    flutterViewController = [[FlutterViewController alloc] initWithEngine:engine
-                                                                  nibName:nil
-                                                                   bundle:nil];
+    return [[FlutterViewController alloc] initWithEngine:engine nibName:nil bundle:nil];
```

Comment:
Can you throw a FLUTTER_ASSERT_ARC or FLUTTER_ASSERT_NOT_ARC at the top of this file?

Reply:
none of the files in testing/ has it. Do you think it's something we'll run into?

Reply:
It's hard to review files if you don't know if they are ARC or not.  You need to import FlutterMacros.h.  The line I've highlighted is correct for ARC, a leak for not ARC.

Reply:
I just meant whether we've had cases where we built the engine locally or on LUCI without ARC in the past? I would imagine everything failing spectacularly everywhere if we did rather than having to gate on a file-per-file basis. But I don't really have much context, I'm just guessing.

Reply:
ARC can be turned on and off on a file-by-file basis.  The Engine is built without ARC, this file is part of a whole different binary though and looks as if it is being compiled with ARC, which is fine.  In a world where some objective-c code is ARC and some is not; it's hard to do code reviews.  The macro is helpful until that magical day that will never come where all of our code is ARC.

Reply:
got it. Thanks for the context. Added